pkg/processor/runtime/rpc/socketallocator.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2024 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rpc
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/common"
+
+	"github.com/nuclio/errors"
+	"github.com/nuclio/logger"
+	"github.com/rs/xid"
+)
+
+const (
+	socketPathTemplate = "/tmp/nuclio-rpc-%s.sock"
+	connectionTimeout  = 2 * time.Minute
+)
+
+type SocketAllocator struct {
+	abstractRuntime *AbstractRuntime
+	logger          logger.Logger
+
+	minSocketsNum        int
+	maxSocketsNum        int
+	eventSockets         []*EventSocket
+	controlMessageSocket *ControlMessageSocket
+}
+
+func NewSocketAllocator(logger logger.Logger, runtime *AbstractRuntime) *SocketAllocator {
+	return &SocketAllocator{logger: logger, minSocketsNum: 1, maxSocketsNum: 1, eventSockets: make([]*EventSocket, 0), abstractRuntime: runtime}
+}
+
+func (sa *SocketAllocator) startListeners() error {
```

Comment:
you don't really start the listeners here but create the socket connections.
Starting would be running the goroutines.